lib/mongoid/core_ext/relations/options.rb

```diff
@@ -1,7 +1,21 @@
 module Mongoid
   module Relations
     module Options
-      COMMON << :versioned
+      VERSIONED_OPTIONS  = [:versioned].freeze
+
+      def validate!(options)
+        valid_options = options[:relation]::VALID_OPTIONS + COMMON + VERSIONED_OPTIONS
```

Comment:
I think it would be less intrusive to use

```
alias_method :_validate!, :validate

def validate!(options)
  return true if options.key?(:versioned)
  _validate!(options)
end
```